lib/daemon_runner/session.rb

```diff
@@ -0,0 +1,124 @@
+require 'diplomat'
+
+module DaemonRunner
+  #
+  # Manage distributed locks with Consul
+  #
+  class Session
+    include Logger
+
+    class << self
+      attr_reader :session
+
+      def start(name, **options)
+        @session ||= Session.new(name, options).renew!
+      end
+
+
+      # Acquire a lock with the current session, or initialize a new session
+      #
+      # @param path [String]  A path in the Consul key-value space to lock
+      # @param lock_session [Session] The Session instance to lock the lock to
+      # @return [Boolean] `true` if the lock was acquired
+      #
+      def lock(path)
+        Diplomat::Lock.acquire(path, session.id)
```

Comment:
If the lock acquire fails it will return false.  It is upto the user to retry the request if they need to.